manager/src/main/java/com/redhat/service/bridge/manager/dao/BridgeDAO.java

```diff
@@ -24,16 +25,16 @@
         return find("#BRIDGE.findByStatusesAndShardId", params).list();
     }
 
-    public Bridge findByNameAndCustomerId(String name, String customerId) {
+    public Optional<Bridge> findByNameAndCustomerId(String name, String customerId) {
```

Comment:
Agreed. @vaibhavjainwiz Lets go for something like `from Bridge b where (b.name=:idOrName or id=:idOrName) and customerId=:customerId` rather than having two queries. That just gives us a single entry point for locating a bridge.

Reply:
done